### ISSUE TITLE
HarmonicMean metric for ignite.metrics

### ISSUE THREAD

**Issue by leelakrishnaRajasimha** [pytorch/ignite#3648]
# Feature
This issue proposes adding a new metric 'HarmonicMean' to ignite.metrics.

The harmonic mean is defined as:

H = n / sum(1 / x_i)

where x_i are the individual values.

This metric can be useful in evaluation scenarios where the harmonic mean of values is required. While Ignite provides several mean-based metrics, there is currently no general-purpose harmonic mean metric.

## Proposed implementation

- A new metric class `HarmonicMean` in `ignite.metrics`
- Supports device placement
- Compatible with distributed training via `sync_all_reduce`
- Includes tests and documentation

Example usage:

```python
metric = HarmonicMean()
metric.attach(evaluator, "harmonic_mean")

**Comment by aaishwarymishra** [comment]
Where is this specific metric is usually used? And is it a common metric?

**Comment by vfdev-5** [comment]
@leelakrishnaRajasimha please provide more context on where this metric is used and how it can be aligned with pytorch-ignite usage/use-cases. Thanks!

**Comment by leelakrishnaRajasimha** [comment]
@vfdev-5 , @aaishwarymishra 
The harmonic mean is commonly used when averaging rates or ratios, where the arithmetic mean can be misleading. A common ML example is the F1-score, which is the harmonic mean of precision and recall.

Ignite already has metrics like Average and Fbeta, but there isn’t a simple general-purpose harmonic mean metric. This metric would allow users to compute harmonic averages of custom values during evaluation while staying consistent with other ignite.metrics.

**Comment by leelakrishnaRajasimha** [comment]
I have also opened a PR#3649 implementing this metric.Please let me know if any changes or improvements are needed.

**Comment by aaishwarymishra** [comment]
@leelakrishnaRajasimha Its usually not a good idea to open pr directly without formal discussion, as the idea could not be accepted.

**Comment by leelakrishnaRajasimha** [comment]
Thanks for the note @aaishwarymishra,
I opened the PR mainly to share a concrete implementation and make it easier to review the idea in context. If the feature itself is not considered useful for Ignite, I'm happy to close the PR or convert it to a draft.

**Comment by vfdev-5** [comment]
> [@vfdev-5](https://github.com/vfdev-5) , [@aaishwarymishra](https://github.com/aaishwarymishra) The harmonic mean is commonly used when averaging rates or ratios, where the arithmetic mean can be misleading. A common ML example is the F1-score, which is the harmonic mean of precision and recall.
> 
> Ignite already has metrics like Average and Fbeta, but there isn’t a simple general-purpose harmonic mean metric. This metric would allow users to compute harmonic averages of custom values during evaluation while staying consistent with other ignite.metrics.

I still do not understand why Ignite should have this metric.
You can code it with current metric features as well:
```python
from ignite.engine import Engine
from ignite.metrics import Average

engine = Engine(lambda e, d: d)

harmonic_metric = Average(output_transform=lambda x: 1.0 / x) ** -1 
harmonic_metric.attach(engine, "harmonic")

data = torch.tensor([1, 2, 3, 4])
state = engine.run(data)
print(state.metrics)
# {'harmonic': 1.9199999908447265}
```

**Comment by leelakrishnaRajasimha** [comment]
Thanks for the example.
My intention was mainly to provide a simple and explicit metric for users, similar to how Average or GeometricMean would be used directly without requiring custom output transforms.
While it can be implemented with Average as shown, having a dedicated HarmonicMean metric could make evaluation code clearer and more consistent with other built-in metrics.
That said, if this is considered unnecessary since it can already be constructed using existing metrics, I am fine closing the PR.

**Comment by vfdev-5** [comment]
I propose the following: if other users ask for the metric, we can reconsider it and move forward with your PR. Right now, you can either keep the PR open in draft mode or close.

**Comment by leelakrishnaRajasimha** [comment]
Thanks for the clarification. I will convert the PR to draft for now. If there is future interest in this metric, we can revisit it.
